modules/acm/src/test/scala/edu/gemini/epics/acm/ObserveStateSpec.scala

```diff
@@ -201,25 +214,28 @@ final class ObserveStateSpec extends FunSuite with MockFactory {
     observe.onCarValChange(CarState.IDLE)
     assert(observe.applyState().isIdle)
 
-    // TODO fix these when the channels are correctly mocked
-    // assert(observeErrorCount.get() === 0)
-    // assert(observePauseCount.get() === 0)
-    // assert(observeSuccessCount.get() === 1)
+    // Check that listener was called
+    assert(observeErrorCount.get() === 0)
+    assert(observePauseCount.get() === 0)
+    assert(observeSuccessCount.get() === 1)
   }
 
   test("GMOS paused observation") {
-    val context: CAJContext = mock[CAJContext]
-    (context.addContextExceptionListener _).expects(*).returns(()).repeat(5)
-    (context.addContextMessageListener _).expects(*).returns(()).repeat(5)
-    (context.pendIO _).expects(*).returns(()).repeat(1 to 6)
-    // We just return null as we don't need the channels and don't want to mock them
-    (context.createChannel(_: String)).expects("gm:apply.DIR").returns(null)
-    (context.createChannel(_: String)).expects("gm:apply.VAL").returns(null)
-    (context.createChannel(_: String)).expects("gm:apply.MESS").returns(null)
-    (context.createChannel(_: String)).expects("gm:applyC.CLID").returns(null)
-    (context.createChannel(_: String)).expects("gm:applyC.VAL").returns(null)
-    (context.createChannel(_: String)).expects("gm:applyC.OMSS").returns(null)
-    val epicsService = new EpicsService(context)
+    val epicsReader = mock[EpicsReader]
+    val epicsWriter = mock[EpicsWriter]
+
+    (epicsWriter.getEnumChannel _).expects("gm:apply.DIR", *).returns(dirChannelPause)
+    (epicsReader.getIntegerChannel _).expects("gm:apply.VAL").returns(intChannelS)
+    (epicsReader.getStringChannel _).expects("gm:apply.MESS").returns(strChannel)
+    (epicsReader.getIntegerChannel _).expects("gm:applyC.CLID").returns(intChannelS)
+    (epicsReader.getEnumChannel _).expects("gm:applyC.VAL", *).returns(carChannel)
+    (epicsReader.getStringChannel _).expects("gm:applyC.OMSS").returns(strChannel)
+    (epicsReader.getIntegerChannel _).expects("gm:dc:observeC.CLID").returns(intChannel)
+    (epicsReader.getEnumChannel _).expects("gm:dc:observeC.VAL", *).returns(carChannel)
+    (epicsReader.getStringChannel _).expects("gm:dc:observeC.OMSS").returns(strChannel)
+    (epicsReader.getShortChannel _).expects("gm:stop.MARK").returns(shortChannel)
+    (epicsReader.getShortChannel _).expects("gm:abort.MARK").returns(shortChannel)
```

Comment:
The same mocks are defined on all the GMOS tests. Can they be factored out ? Same for all other instruments.